etc/pre-commit.sh

```diff
@@ -0,0 +1,30 @@
+#! /usr/bin/env bash
```

Comment:
I think it's a fine idea to have a pre-commit hook around that people can install if they want to; I've never been a user of them myself but it's nice to have the option.
that said, I think for maintainability / any future customization other users might want to do, it would be useful to add more comments in here explaining how it works. for example, as a casual reader I'm not totally sure why FILES_CHANGED is being determined in that way with the filter and all